proto/src/lib.rs

```diff
@@ -93,7 +93,7 @@ impl Decoder for LdapCodec {
         }
         // Build the LdapMsg from the Tag
         LdapMsg::try_from(msg)
-            .map_err(|_| io::Error::new(io::ErrorKind::Other, "ldapmsg invalid"))
+            .map_err(|e| io::Error::new(io::ErrorKind::Other, format!("ldapmsg invalid: {:?}", e)))
```

Comment:
Nitpick: [`io::Error::other`](https://doc.rust-lang.org/std/io/struct.Error.html#method.other) method has been stabilized in Rust 1.74 (depending on the msrv policy for the crate, you may use it)

It’s also possible to just construct the io::Error while keeping the original error like this:
```rust
            .map_err(io::Error::other)
```
which is equivalent to
```rust
            .map_err(|e| io::Error::other(e))
```

However, you lose the "ldapmsg invalid" bit.

Reply:
We don't mind bumping the msrv up if you want to use this feature.